File: sort-array-by-parity-ii/sort-array-by-parity-ii.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        int n=nums.size();
        vector<int> res;
        vector<int> e;
        vector<int> o;
        for(int i=0;i<n;i++){
            if(nums[i]%2==0){
                e.push_back(nums[i]);
            }
            else
                o.push_back(nums[i]);
        }
        int ei=0,oi=0;
        for(int i=0;i<n;i++){
            if(i%2)
                res.push_back(o[ei++]);
            else
                res.push_back(e[oi++]);
        }
        return res;
    }
};

File: sort-array-by-parity-ii/sort-array-by-parity-ii.java
class Solution {
    public int[] sortArrayByParityII(int[] nums) {
        int i=0,j=1;
        int n=nums.length;
        while(i<n && j<n){
            if(nums[i]%2==0)
                i+=2;
            else if(nums[j]%2==1)
                j+=2;
            else{
                int temp=nums[i];
                nums[i]=nums[j];
                nums[j]=temp;
            }
        }
        return nums;
    }
}